apps/web/src/ee/clerk/providers/EnterpriseAuthProvider.tsx

```diff
@@ -147,27 +147,55 @@ export const EnterpriseAuthProvider = ({ children }: { children: React.ReactNode
   return <EnterpriseAuthContext.Provider value={value}>{children}</EnterpriseAuthContext.Provider>;
 };
 
-const toUserEntity = (clerkUser: UserResource): IUserEntity => ({
-  _id: clerkUser.id,
-  firstName: clerkUser.firstName,
-  lastName: clerkUser.lastName,
-  email: clerkUser.emailAddresses[0].emailAddress,
-  profilePicture: clerkUser.imageUrl,
-  createdAt: clerkUser.createdAt?.toString() ?? '',
-  showOnBoarding: clerkUser.publicMetadata.showOnBoarding,
-  showOnBoardingTour: clerkUser.publicMetadata.showOnBoardingTour,
-  servicesHashes: clerkUser.publicMetadata.servicesHashes,
-  jobTitle: clerkUser.publicMetadata.jobTitle,
-  hasPassword: clerkUser.passwordEnabled,
-});
-
-const toOrganizationEntity = (clerkOrganization: OrganizationResource): IOrganizationEntity => ({
-  _id: clerkOrganization.id,
-  name: clerkOrganization.name,
-  createdAt: clerkOrganization.createdAt.toString(),
-  updatedAt: clerkOrganization.updatedAt.toString(),
-  apiServiceLevel: clerkOrganization.publicMetadata.apiServiceLevel,
-  defaultLocale: clerkOrganization.publicMetadata.defaultLocale,
-  domain: clerkOrganization.publicMetadata.domain,
-  productUseCases: clerkOrganization.publicMetadata.productUseCases,
-});
+const toUserEntity = (clerkUser: UserResource): IUserEntity => {
+  /*
+   * When mapping to IUserEntity, we have 2 cases:
+   *  - user exists and has signed in
+   *  - user is signing up
+   *
+   * In the case where the user is still signing up, we are using the clerk identifier for the id.
+   * This however quickly gets update to the externalId (which is actually the novu internal entity
+   * identifier) that gets used further in the app. A few consumers that want to user this identifier
+   * before it is set to the internal value monitor the flag 'shouldMonitor' and only use them when
+   * the internal Novu id is present.
+   */
+
+  return {
+    _id: clerkUser.externalId ?? clerkUser.id,
+    firstName: clerkUser.firstName,
+    lastName: clerkUser.lastName,
+    email: clerkUser.emailAddresses[0].emailAddress,
+    profilePicture: clerkUser.imageUrl,
+    createdAt: clerkUser.createdAt?.toString() ?? '',
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** The mapper below doesn't include the nullish coalescing operator (`??`) for the `createdAt`, is it really necessary here? When would a `clerkUser` exist without a `createdAt` property?‏

Reply:
that is a good question, i will check in the docs and remove if it isn't needed

Reply:
Indeed, there is a difference in the types, for the `UserResource` it can be set to `null`. I've asked the Clerk guys if they have an explanation for this.